ansible/library/bgp_route.py

```diff
@@ -0,0 +1,286 @@
+        return
+
+    def get_facts(self):
+        return self.facts
+
+    def parse_bgp_rt_adv(self, cmd, cmd_result):
+        '''
+        parse BGP routing facts of neighbor advertised routes
+        '''
+        regex_neighbor = re.compile('nei.*\s([0-9a-fA-F.:]+)\s.*')
+        #regix_iprange = re.compile('[0-9a-fA-F.:]+\/\d+')
+        if regex_neighbor.match(cmd):
+            neighbor = regex_neighbor.match(cmd).group(1)
+        else:
+            raise Exception('cannot find neighbor in "show ip  bgp ' + cmd + '"')
+        self.facts['bgp_rt_neiadv']['neighbor'] = neighbor
+        ### so far parsing by fixed length table
+        ### will improve this if find better way to parse it
+        header_position = [0, 3, 20, 40, 47, 54, 61]
+        header = 'Network          Next Hop            Metric LocPrf Weight Path'
+        result_lines = cmd_result.split('\n')
+        table_start = False
+        while len(result_lines) != 0:
+            line = result_lines.pop(0)
+            if not table_start:
+                if header in line:
+                    table_start = True
+                    continue
+            else:
+                if not re.match('^\*', line):
+                    continue
+                entry = {}
+                if len(line) > 30:   ## route entry in one line
+                    prefix = line[header_position[1]:header_position[2]].strip()
+                else:    ### route entry in two lines
+                    prefix = line[header_position[1]:30].strip()
+                    line = result_lines.pop(0)
+                entry['next_hop'] = line[header_position[2]:header_position[3]].strip()
+                paths = line[header_position[6]:].strip()
+                entry['paths'] = re.findall('[0-9]+', paths)
+                self.facts['bgp_rt_neiadv'][prefix] = entry
+
+
+    def parse_bgp_rt_prefix(self, prefix, cmd_result):
+        '''
+        parse BGP facts for specific prefix
+        '''
+        regex_prefix_header = re.compile('BGP routing table entry for ')
+        regex_prefix_avail_paths = re.compile('Paths:\s+\((\d+)\s+available')
+        prefix_temp_line1 = 'Advertised to non peer-group peers:'
+        regex_prefix_prop2 = re.compile('^[0-9a-fA-F.:\s]+$')
+        regex_prefix_paths = re.compile('^[0-9\s]+$')
+        regex_prefix_path_line1 = re.compile('.* from .*\([0-9a-fA-F.:]+\)')
+        regex_prefix_path_line2 = re.compile('\s+Origin')
+        regex_prefix_path_line3 = re.compile('\s+Last update:')
+        cmd_err1 = 'Unknown command'
+        cmd_err2 = 'Network not in table'
+        self.facts['bgp_rt'] = defaultdict(dict)
+        if cmd_err1 in cmd_result or cmd_err2 in cmd_result:
+            self.facts['bgp_rt'][prefix]['found'] = False
+            return
+        result_lines = cmd_result.split('\n')
+        ## search_state = ['header', 'path_prop', 'prefix_prop1', 'prefix_prop2', 'paths', 'path_l1', 'path_l2', 'path_l3', 'err' ]
```

Comment:
>## search_state = ['header [](start = 8, length = 26)

you seems like to like to compare strings, but it is better to define constant and then use the contant

for example, 
HEADER_LINE=1
XXX_LINE=2
...